### ISSUE TITLE
📎 Implement `noUselessStringConcat` - `no-useless-concat`

### ISSUE THREAD

**Issue by michellocana** [biomejs/biome#2622]
### Description

[no-useless-concat](https://eslint.org/docs/latest/rules/no-useless-concat).

I created this issue just to keep track of my progress on the rule implementation, as [discussed here](https://github.com/biomejs/biome/discussions/3#discussioncomment-9241628). The idea is to prevent useless string concatenations.

## Valid
```js
const a = 1 + 1;
const a = 1 * '2';
const a = 1 - 2;
const a = foo + bar;
const a = 'foo' + bar;
const a = (number + 1) + 'px'
const a = 'a' + 1
const a = 1 + '1'
const a = 1 + 1 + `1`
const a = `1` + 1
const a = (1 + +2) + `b`
```
## Invalid
```js
const a = 'a' + 'b'
const a = 'a' + 'b' + 'c'
const a = foo + 'a' + 'b'
const a = (foo + 'a') + ('b' + 'c')
const a = `a` + 'b'
const a = `a` + `b`
const a = foo + `a` + `b`
```

I got these examples from the equivalent ESLint rule tests, but I'm not entirely sure if some of the cases shouldn't be considered invalid, like `1 + '1'` or `'a' + 1`, or maybe the ESLint rule idea is to not change the type of the values? I'm open to discussions :)
